clients/src/main/java/org/apache/kafka/common/security/authenticator/CredentialCache.java

```diff
@@ -16,13 +16,12 @@
  */
 package org.apache.kafka.common.security.authenticator;
 
-import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class CredentialCache {
 
-    private final Map<String, Cache<? extends Object>> cacheMap = new HashMap<>();
+    private final Map<String, Cache<? extends Object>> cacheMap = new ConcurrentHashMap<>();
 
     public <C> Cache<C> createCache(String mechanism, Class<C> credentialClass) {
```

Comment:
We don't really need this to be a ConcurrentHashMap since we create all the caches on broker startup, but changed to `putIfAbsent` just in case we change that in future.